### ISSUE TITLE
broot closes and everything it called with it (sometimes not though... ??) as the tilling wm resizes the window too short

### ISSUE THREAD

**Issue by evanescente-ondine** [Canop/broot#511]
Hi,
I wonder why, I only run into that issue now.
Before, when the new window size would get too small, th terminal would write "no enough space" or something like that, broot being inaccessible and the terminal unusable even with size restored.
But it wouldn't close, nor the files opened. And since we can't open things in background yet... yeah
I don't know what prompted the change in behavior.
Ideally I would like the window not close but broot refreshing when the size is sufficient once again, or  background jobs, or broot refusing the WM's dimensions change, or asking to go floating. Any one is good.

Thanks

**Comment by Canop** [comment]
I don't think it ever changed for broot, but I could do what I do in a few other TUI applications: just don't render the main areas when there's not enough room.

**Comment by evanescente-ondine** [comment]
I think its requirement in vertical space is too severe.
Moreover, I think it should refuse opening if it decides it's too short. what happens, is that it opens, doesn't like the dimension, then shuts down down, but the window it forced the WM to resize also gets to shutdown.
it happened several times for me because I open broot for whatever reason... copying filename (no need for "basename" invocation anymore !), launcher.. hardly any use for a shell now. My dream is coming true: an interface with shell and programs fully integrated...
In this picture, I think the smaller broots should allow for half the vertical space. Some people here have small screens...
![ici](https://imghost.org/images/2022/02/21/error_broot.md.jpg)

So I would rather do without that less than predictable closing of already opened instances.

**Comment by Canop** [comment]
> the window it forced the WM to resize also gets to shutdown

Hu ? You mean the terminal breaks when broot quits in case of too small size ?

**Comment by evanescente-ondine** [comment]
yes.
Sequence of events
1) Instance 1 is opened and fine.
2) instance 2 opens to a small size, and instance 1 is resized to about the same dimensions as instance 2.
3) Immediately, both being too small the two windows close.

**Comment by Canop** [comment]
broot, starting from 1.10.0, doesn't close anymore when the terminal is too small.
